tests/optimizers/conftest.py

```diff
@@ -8,63 +8,109 @@
+    pso = GlobalBestPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5})
     pso.optimize(sphere_func, 1000, verbose=0)
     return pso
 
-@pytest.fixture(scope='module')
+
+@pytest.fixture(scope="module")
 def gbest_reset():
     """Returns a GlobalBestPSO instance that has been run and reset to check
     default value"""
-    pso = GlobalBestPSO(10, 2, {'c1': 0.5, 'c2': 0.7, 'w': 0.5})
+    pso = GlobalBestPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5})
     pso.optimize(sphere_func, 10, verbose=0)
     pso.reset()
     return pso
 
-@pytest.fixture(scope='module')
+
+@pytest.fixture(scope="module")
 def lbest_history():
     """Returns a LocalBestPSO instance run for 1000 iterations for checking
     history"""
-    pso = LocalBestPSO(10, 2, {'c1': 0.5, 'c2': 0.7, 'w': 0.5, 'k':2, 'p':2})
+    pso = LocalBestPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5, "k": 2, "p": 2})
     pso.optimize(sphere_func, 1000, verbose=0)
     return pso
 
-@pytest.fixture(scope='module')
+
+@pytest.fixture(scope="module")
 def lbest_reset():
     """Returns a LocalBestPSO instance that has been run and reset to check
     default value"""
-    pso = LocalBestPSO(10, 2, {'c1': 0.5, 'c2': 0.7, 'w': 0.5, 'k': 2, 'p': 2})
+    pso = LocalBestPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5, "k": 2, "p": 2})
     pso.optimize(sphere_func, 10, verbose=0)
     pso.reset()
     return pso
 
-@pytest.fixture(scope='module')
+
+@pytest.fixture(scope="module")
 def binary_history():
     """Returns a BinaryPSO instance run for 1000 iterations for checking
     history"""
-    pso = BinaryPSO(10, 2, {'c1': 0.5, 'c2': 0.7, 'w': 0.5, 'k':2, 'p':2})
+    pso = BinaryPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5, "k": 2, "p": 2})
     pso.optimize(sphere_func, 1000, verbose=0)
     return pso
 
-@pytest.fixture(scope='module')
+
+@pytest.fixture(scope="module")
 def binary_reset():
     """Returns a BinaryPSO instance that has been run and reset to check
     default value"""
-    pso = BinaryPSO(10, 2, {'c1': 0.5, 'c2': 0.7, 'w': 0.5, 'k': 2, 'p': 2})
+    pso = BinaryPSO(10, 2, {"c1": 0.5, "c2": 0.7, "w": 0.5, "k": 2, "p": 2})
     pso.optimize(sphere_func, 10, verbose=0)
     pso.reset()
     return pso
 
+
 @pytest.fixture
 def options():
     """Default options dictionary for most PSO use-cases"""
-    options_ = {'c1':0.5, 'c2':0.7, 'w':0.5, 'k':2, 'p':2}
-    return options_
\ No newline at end of file
+    options_ = {"c1": 0.5, "c2": 0.7, "w": 0.5, "k": 2, "p": 2}
+    return options_
+
+
+@pytest.fixture(params=[
+                Star(),
+                Ring(static=False), Ring(static=True),
+                Pyramid(static=False), Pyramid(static=True),
+                Random(static=False), Random(static=True)
+                ])
+def topology(request):
+    """Parametrized topology parameter"""
+    topology_ = request.param
+    return topology_
+
+
+@pytest.fixture(scope="module", params=[True, False])
+def static(request):
```

Comment:
To be consistent we should replace this fixture with parametrization marks.

Reply:
I wonder why this showed up again, didn't I comment on this before? Or is it in a different `conftest`?

Reply:
It's in a different `conftest` 😄